system/settings/settings.xml

```diff
@@ -1710,17 +1710,19 @@
           </updates>
           <control type="edit" format="integer" />
         </setting>
+        <setting id="services.webserverauthentication" parent="services.webserver" type="boolean" label="1052" help="1053">
+          <level>2</level>
+          <default>true</default>
+          <control type="toggle" />
+        </setting>
         <setting id="services.webserverusername" type="string" parent="services.webserver" label="1048" help="36330">
           <level>1</level>
           <default>kodi</default>
-          <constraints>
-            <allowempty>true</allowempty>
-          </constraints>
           <updates>
             <update type="change" />
           </updates>
           <dependencies>
-            <dependency type="enable" setting="services.webserver">true</dependency>
+            <dependency type="enable" setting="services.webserverauthentication">true</dependency>
```

Comment:
This is intentional, actually. So that you can set up username and password before actually starting the server. Similarly to the port setting which is also enabled (editable) when `services.webserver` is disabled. Or do you think this is a problem?

Reply:
No if it is intentional it's OK. I didn't consider this use case.